app/themes/default/components/ui/storefront/ProductCard.tsx

```diff
@@ -38,7 +38,12 @@ const ProductCard = ({
           <p className="mt-3 text-lg">{price}</p>
         </CardFooter>
       </Link>
-      <Button variant="secondary" className="w-full" onClick={onClick}>
+      <Button
+        variant="secondary"
+        className="w-full"
+        onClick={onClick}
+        data-testid={id}
```

Comment:
the data-testid will work like ref?

Reply:
it's used by react-testing-library to anchor a test target, FYI https://testing-library.com/docs/queries/bytestid/